File: CS/PS/Site/programmers/DFS BFS/타겟넘버.js
function solution(numbers, target) {
  let answer = 0;
  
  const dfs = (numbers, target, sum, index) => {
    if (numbers.length <= index){
      if (sum === target) answer++;
      return;
    }
    dfs(numbers, target, sum + numbers[index], index + 1);
    dfs(numbers, target, sum - numbers[index], index + 1);

  }
  dfs(numbers, target, 0, 0);
  return answer;
}


console.log(solution([1,1,1,1,1], 3));

File: CS/PS/Site/programmers/DFS BFS/타겟넘버.py
answer = 0

def dfs(numbers, target, sum, index):
  global answer
  length = len(numbers)
  if (length <= index):
    if (sum == target):
      answer += 1
    return ;
  dfs(numbers, target, sum + numbers[index], index + 1)
  dfs(numbers, target, sum - numbers[index], index + 1)

def solution(numbers, target):
  global answer
  dfs(numbers, target, 0, 0)
  print(answer)
  return answer